### DISCUSSION TITLE
Something about Data format exporting and operation on matrix

### DISCUSSION THREAD

**Discussion by jefferyUstc** [GreenleafLab/ArchR#611] [Questions / Documentation]
Hi, I have gone through the paper and docs these days and great work! And I have some questions in my utilization on my own data analysis when using ArchR.

1. are there any utility interfaces to export h5ad or Seurat formats?
2. or instead, how can i get access to the matrixs in getAvailableMatrices(proj) . i hope i can get and set custom matrix. for example, set my own peakmatrix  called from outside.

thank you so much!

**Comment by rcorces** [comment]
Thanks for using ArchR!

> 1. are there any utility interfaces to export h5ad or Seurat formats?

This is not explicitly possible in ArchR. If you find yourself wishing ArchR had some functionality that exists in other packages (for ex. Signac) feel free to open a Discussion thread for a feature request. ArchR does allow you to export data in some ways, mostly by creating `SummarizedExperiments` objects. This is illustrated at various points in the [manual](https://www.archrproject.com/bookdown/index.html)

> 2. how can i get access to the matrixs in getAvailableMatrices(proj) . i hope i can get and set custom matrix. for example, set my own peakmatrix called from outside.

For export of a matrix - I believe you are looking for the `getMatrixFromProject()` function. https://www.archrproject.com/reference/getMatrixFromProject.html
For import - you cannot import matrices yet but you can import your own peak set defined outside of ArchR and use this to create a peak matrix within ArchR. see `addPeakSet()` https://www.archrproject.com/reference/addPeakSet.html

If this answers your questions, feel free to mark this as "Answered"

**Reply by rcorces** [reply]
I would recommend familiarizing yourself with what a `SummarizedExperiment` object is and how to work with one.
https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html

**Reply by brianpenghe** [reply]
what about exporting into a Seurat object?

**Reply by rcorces** [reply]
This isnt currently possible.

**Reply by brianpenghe** [reply]
I think one thing I can do is to save matrices from the ArchR object into matrix market formats and then read into other platforms
